src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -951,48 +923,36 @@ export async function getFills(
   return fills;
 }
 
-/**
- * @description Search for refund requests recorded by a specific SpokePool.
- * @param chainId Chain ID of the relevant SpokePoolClient instance.
- * @param chainIdIndices Complete set of ordered chain IDs.
- * @param hubPoolClient HubPoolClient instance.
- * @param spokePoolClients Set of SpokePoolClient instances, mapped by chain ID.
- * @param filter  Optional filtering criteria.
- * @returns Array of RefundRequestWithBlock events matching the chain ID and optional filtering criteria.
- */
-export async function getRefundRequests(
+export async function getValidRefundCandidates(
   chainId: number,
   hubPoolClient: HubPoolClient,
   spokePoolClients: SpokePoolClients,
-  filter: SpokePoolEventFilter = {}
-): Promise<RefundRequestWithBlock[]> {
+  filter: Pick<SpokePoolFillFilter, "fromBlock" | "toBlock"> = {}
+): Promise<(RefundRequestWithBlock & { matchedDeposit: DepositWithBlock })[]> {
   const spokePoolClient = spokePoolClients[chainId];
   assert(isDefined(spokePoolClient));
 
-  const { originChainId, destinationChainId, relayer, fromBlock } = filter;
-
-  const refundRequests = await filterAsync(spokePoolClient.getRefundRequests(), async (refundRequest) => {
-    assert(refundRequest.repaymentChainId === chainId);
-
-    if (isDefined(fromBlock) && fromBlock > refundRequest.blockNumber) {
-      return false;
-    }
+  const { fromBlock, toBlock } = filter;
 
-    // @dev tsdx and old Typescript seem to prevent dynamic iteration over the filter, so evaluate the keys manually.
-    if (
-      (isDefined(originChainId) && refundRequest.originChainId !== originChainId) ||
-      (isDefined(destinationChainId) && refundRequest.destinationChainId !== destinationChainId) ||
-      (isDefined(relayer) && refundRequest.relayer !== relayer)
-    ) {
-      return false;
-    }
-
-    const result = await refundRequestIsValid(spokePoolClients, hubPoolClient, refundRequest);
-
-    return result.valid;
-  });
```

Comment:
Can we retain this filtering? It's optional to use, and the relayer relies on filtering by `relayer` to find its existing refund requests.

Reply:
sure